test/unit-tests.c

```diff
@@ -348,12 +348,21 @@ test_realize_ctx(void **state __attribute__((unused)))
 static void
 test_attach_ctx(void **state __attribute__((unused)))
 {
+    int dummy_attach(vfu_ctx_t *vfu_ctx)
+    {
+        assert(vfu_ctx != NULL);
```

Comment:
What I meant was to define `vfu_ctx` before `dummy_attach` , and then check in `dummy_attach` that the passed argument contains the correct value, .e.g

```
...
vfu_ctx_t vfu_ctx;

int dummy_attach(vfu_ctx_t *_vfu_ctx) {
    assert_ptr_equal(_vfu_ctx == &vfu_ctx);
    return 0xdeadbeef;
}
...
```
Anyway that's minor stuff.